pandas/core/groupby/grouper.py

```diff
@@ -348,9 +350,19 @@ def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
             if getattr(self.grouper, "name", None) == key and isinstance(
                 obj, ABCSeries
             ):
-                # pandas\core\groupby\grouper.py:348: error: Item "None" of
-                # "Optional[Any]" has no attribute "take"  [union-attr]
-                ax = self._grouper.take(obj.index)  # type: ignore[union-attr]
+                # Sometimes self._grouper will have been resorted while
```

Comment:
can you track down *exactly* how this happens, this is a real problem, somthing is keeping state and now we are undoing it. Your fix is pretty clean, but this makes reasoning very about this very hard.

Reply:
It’s very complicated and you have to dig through a stack trace that is 18 method calls deep to understand it but I believe it’s because self._grouper is meant to be the TimeGrouper though in this case it’s actually just a sorted DatetimeIndex and obj if you go up the stack trace you’ll see that it is coming from a SeriesGroupby._selected_obj which I’m guessing just doesn’t get sorted when it’s created.  The case where self._grouper has not been resorted I believe only happens when the dates were already in order and don’t need to be resorted.